stripe/src/main/java/com/stripe/android/StripeApiHandler.java

```diff
@@ -702,84 +472,134 @@ private static void fireAndForgetApiCall(
+            return flatParams;
         }
 
-        StripeResponse response = getStripeResponse(method, url, params, options);
+        for (Map.Entry<String, Object> entry : params.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
 
-        int rCode = response.getResponseCode();
-        String rBody = response.getResponseBody();
+            String newPrefix = key;
+            if (keyPrefix != null) {
+                newPrefix = String.format("%s[%s]", keyPrefix, key);
+            }
 
-        String requestId = null;
-        Map<String, List<String>> headers = response.getResponseHeaders();
-        List<String> requestIdList = headers == null ? null : headers.get("Request-Id");
-        if (requestIdList != null && requestIdList.size() > 0) {
-            requestId = requestIdList.get(0);
+            flatParams.addAll(flattenParamsValue(value, newPrefix));
         }
 
-        if (rCode < 200 || rCode >= 300) {
-            handleAPIError(rBody, rCode, requestId);
+        return flatParams;
+    }
+
+    private static List<Parameter> flattenParamsValue(Object value, String keyPrefix)
+            throws InvalidRequestException {
+        List<Parameter> flatParams;
+
+        if (value instanceof Map<?, ?>) {
+            flatParams = flattenParamsMap((Map<String, Object>) value, keyPrefix);
+        } else if (value instanceof List<?>) {
+            flatParams = flattenParamsList((List<Object>) value, keyPrefix);
+        } else if ("".equals(value)) {
+            throw new InvalidRequestException("You cannot set '"+keyPrefix+"' to an empty string. "+
+                    "We interpret empty strings as null in requests. "+
+                    "You may set '"+keyPrefix+"' to null to delete the property.",
+                    keyPrefix, null, 0, null);
+        } else if (value == null) {
+            flatParams = new LinkedList<>();
+            flatParams.add(new Parameter(keyPrefix, ""));
+        } else {
+            flatParams = new LinkedList<>();
+            flatParams.add(new Parameter(keyPrefix, value.toString()));
         }
 
-        if (allowedToSetTTL) {
-            if (originalDNSCacheTTL == null) {
-                // value unspecified by implementation
-                // DNS_CACHE_TTL_PROPERTY_NAME of -1 = cache forever
-                java.security.Security.setProperty(
-                        DNS_CACHE_TTL_PROPERTY_NAME, "-1");
-            } else {
-                java.security.Security.setProperty(
-                        DNS_CACHE_TTL_PROPERTY_NAME, originalDNSCacheTTL);
-            }
+        return flatParams;
+    }
+
+    private static String formatURL(String url, String query) {
+        if (query == null || query.isEmpty()) {
+            return url;
+        } else {
+            // In some cases, URL can already contain a question mark (eg, upcoming invoice lines)
+            String separator = url.contains("?") ? "&" : "?";
```

Comment:
Yeah - a good deal of this was generously sampled from stripe-java's [LiveStripeResponseGetter](https://github.com/stripe/stripe-java/blob/master/src/main/java/com/stripe/net/LiveStripeResponseGetter.java) and not adjusted too much in the mean time.